pkg/ansible/controller/reconcile.go

```diff
@@ -185,31 +185,42 @@ func (r *AnsibleOperatorReconciler) Reconcile(request reconcile.Request) (reconc
 			}
 		}
 		u.SetFinalizers(finalizers)
-		needsUpdate = true
-	}
-
-	statusMap, ok := u.Object["status"].(map[string]interface{})
-	if !ok {
-		u.Object["status"] = ResourceStatus{
-			Status: NewStatusFromStatusJobEvent(statusEvent),
-		}
-		logrus.Infof("adding status for the first time")
-		needsUpdate = true
-	} else {
-		// Need to conver the map[string]interface into a resource status.
-		if update, status := UpdateResourceStatus(statusMap, statusEvent); update {
-			u.Object["status"] = status
-			needsUpdate = true
+		err := r.Client.Update(context.TODO(), u)
+		if err != nil {
+			return reconcileResult, err
 		}
 	}
-	if needsUpdate {
-		err = r.Client.Update(context.TODO(), u)
-	}
+	ansibleStatus := ansiblestatus.NewAnsibleResultFromStatusJobEvent(statusEvent)
+
 	if !runSuccessful {
-		reconcileResult.Requeue = true
-		return reconcileResult, err
-	}
+		sc := ansiblestatus.GetCondition(crStatus, ansiblestatus.RunningConditionType)
+		sc.Status = v1.ConditionFalse
+		ansiblestatus.SetCondition(&crStatus, *sc)
+		c := ansiblestatus.NewCondition(
+			ansiblestatus.FailureConditionType,
+			v1.ConditionTrue,
+			ansibleStatus,
+			ansiblestatus.FailedReason,
+			strings.Join(failureMessages, "\n"),
+		)
+		ansiblestatus.SetCondition(&crStatus, *c)
+	} else {
+		c := ansiblestatus.NewCondition(
+			ansiblestatus.RunningConditionType,
+			v1.ConditionTrue,
+			ansibleStatus,
+			ansiblestatus.SuccessfulReason,
+			ansiblestatus.SuccessfulMessage,
```

Comment:
Would it be useful to record what you did to some degree in the message? Or perhaps you just want that to be in the logs/events?